test/validation/validation.go

```diff
@@ -0,0 +1,236 @@
+		OperatorNameSpace = ns
+	}
+
+	junitPath = flag.String("junit", "", "the path for the junit format report")
+	reportPath = flag.String("report", "", "the path of the report file containing details for failed tests")
+}
+
+func RunValidationTests(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	rr := []Reporter{}
+	if *junitPath != "" {
+		junitFile := path.Join(*junitPath, "e2e_junit.xml")
+		rr = append(rr, reporters.NewJUnitReporter(junitFile))
+	}
+
+	clients := testclient.New("")
+
+	if *reportPath != "" {
+		rr = append(rr, k8sreporter.New(clients, OperatorNameSpace, *reportPath))
+	}
+
+	RunSpecsWithDefaultAndCustomReporters(t, "Metallb Operator Validation Suite", rr)
+}
+
+var _ = Describe("metallb", func() {
+	Context("Platform Check", func() {
+		It("Should have the MetalLB Operator namespace", func() {
+			_, err := testclient.Client.Namespaces().Get(context.Background(), OperatorNameSpace, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred(), "Should have the MetalLB Operator namespace")
+		})
+		It("should be either Kubernetes or OpenShift platform", func() {
+			cfg := ctrl.GetConfigOrDie()
+			platforminfo, err := platform.GetPlatformInfo(cfg)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(platforminfo.IsOpenShift()).Should(Equal(TestIsOpenShift))
+		})
+	})
+
+	Context("MetalLB", func() {
+		It("should have the MetalLB Operator deployment in running state", func() {
+			Eventually(func() bool {
+				deploy, err := testclient.Client.Deployments(OperatorNameSpace).Get(context.Background(), consts.MetalLBOperatorDeploymentName, metav1.GetOptions{})
+				if err != nil {
+					return false
+				}
+				return deploy.Status.ReadyReplicas == deploy.Status.Replicas
+			}, util.Timeout, util.Interval).Should(BeTrue())
+
+			pods, err := testclient.Client.Pods(OperatorNameSpace).List(context.Background(), metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("control-plane=%s", consts.MetalLBOperatorDeploymentLabel)})
+			Expect(err).ToNot(HaveOccurred())
+
+			deploy, err := testclient.Client.Deployments(OperatorNameSpace).Get(context.Background(), consts.MetalLBOperatorDeploymentName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(pods.Items)).To(Equal(int(deploy.Status.Replicas)))
+
+			for _, pod := range pods.Items {
+				Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
+			}
+		})
+
+		It("should have the MetalLB CRD available in the cluster", func() {
+			crd := &apiext.CustomResourceDefinition{}
+			err := testclient.Client.Get(context.Background(), goclient.ObjectKey{Name: consts.MetalLBOperatorCRDName}, crd)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should have the MetalLB AddressPool CRD available in the cluster", func() {
+			crd := &apiext.CustomResourceDefinition{}
+			err := testclient.Client.Get(context.Background(), goclient.ObjectKey{Name: consts.MetalLBAddressPoolCRDName}, crd)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("MetalLB deploy", func() {
```

Comment:
doesn't this belong to e2e?

Reply:
I was having a dilema here, the .Create() does not look good in validation tests indeed. Moved